flaml/autogen/agentchat/responsive_agent.py

```diff
@@ -260,19 +260,19 @@ def _append_oai_message(self, message: Union[Dict, str], role, conversation_id:
 
         Args:
             message (dict or str): message to be appended to the ChatCompletion conversation.
-            role (str): role of the message, can be "assistant" or "function".
+            role (str): role of the message, can be "assistant", "user" or "function", this will overwrite the `role` in the message.
             conversation_id (Agent): id of the conversation, should be the recipient or sender.
 
         Returns:
             bool: whether the message is appended to the ChatCompletion conversation.
         """
         message = self._message_to_dict(message)
         # create oai message to be appended to the oai conversation that can be passed to oai directly.
-        oai_message = {k: message[k] for k in ("content", "function_call", "name", "context") if k in message}
-        if "content" not in oai_message and "function_call" not in oai_message:
+        oai_message = {k: message[k] for k in ("content", "function_call", "name", "role") if k in message}
+        # 'content' is a required field
+        if "content" not in oai_message or role not in ("assistant", "user", "function"):
             return False
-
-        oai_message["role"] = "function" if message.get("role") == "function" else role
+        oai_message["role"] = role
```

Comment:
when the role is "user" and oai_message["role"]="function", do we want to overwrite oai_message["role"]?

Reply:
Yes. One purpose of this function is to change roles for some messages
However, the "role" is set as an required argument now, I think we can set it to optional and overwrite only when it is passed in.

Reply:
Then we need to change the part where this function is called. Otherwise, with your change, the function call feature will not work.